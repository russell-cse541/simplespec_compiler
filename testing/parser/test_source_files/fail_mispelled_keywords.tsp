// The word requires is misspelled on line 13.
// The word const is misspelled on line 27.

namespace  Main;

using Locale;
using Big;

scalar ID = ASCIIString {
    requires: value.length();
}

scalar Apple = ASCIIString {
    requiress: component.juice();
    requires: component.seed();
}


model User {
    id: ID;
    name: String;
    region: Locale::Country;
    team: List<User,Hey>;
    fan: Bad<Yes>;
}

cost a:Bool = true;
const sillString:String = 'Yes This is a very Simple String!';

function checkit(): Bool {
    let b:Bool = true;
    let a:String = a.length();
    return u.a.length(pred(a:String): Bool => true);
}

function checkit(a:String, b:Bool, c:SomethingFake): Bool {
    return !(5>8) || (!(a || b && c));
}