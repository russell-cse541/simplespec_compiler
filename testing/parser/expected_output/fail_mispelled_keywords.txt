Error: syntax error line: 14
Error: [1;31m Improper Model/Scalar [0m line: 14
Error: syntax error line: 27
Error: [1;31m Improper Declaration [0m line: 27
Error: syntax error line: 28
Error: [1;31m Improper Declaration [0m line: 28
Error: syntax error line: 30
Error: [1;31m Improper Declaration [0m line: 31
Error: syntax error line: 32
Error: [1;31m Improper Declaration [0m line: 32
Error: syntax error line: 33
Error: [1;31m Improper Declaration [0m line: 33
Error: syntax error line: 34
Error: [1;31m Improper Declaration [0m line: 37
Error: syntax error line: 38


[0;33m There were errors parsing the document. Start by fixing the initial error.  Other errors may be uncovered as some are corrected.[0m

